File: server/src/main/java/kaigee/top/user/root/controller/UserForFrontController.java
package kaigee.top.user.root.controller;
import cn.dev33.satoken.annotation.SaCheckDisable;
import cn.dev33.satoken.stp.SaTokenInfo;
import kaigee.top.infrastructure.model.QueryRequest;
import kaigee.top.user.root.entity.User;
import kaigee.top.user.root.entity.dto.UserInput;
import kaigee.top.user.root.entity.dto.UserLoginInput;
import kaigee.top.user.root.entity.dto.UserSpec;
import kaigee.top.user.root.repository.UserRepository;
import kaigee.top.user.root.service.UserService;
import lombok.AllArgsConstructor;
import org.babyfish.jimmer.client.FetchBy;
import org.babyfish.jimmer.client.meta.DefaultFetcherOwner;
import org.springframework.data.domain.Page;
import org.springframework.transaction.annotation.Transactional;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("front/user")
@AllArgsConstructor
@DefaultFetcherOwner(UserRepository.class)
@Transactional
@SaCheckDisable
public class UserForFrontController {

    private final UserService userService;

    @GetMapping("{id}")
    public @FetchBy(value = "USER_ROLE_FETCHER") User findById(@PathVariable String id) {
        return userService.findById(id);
    }

    @PostMapping("query")
    public Page< @FetchBy(value = "COMPLEX_FETCHER") User> query(@RequestBody QueryRequest<UserSpec> queryRequest) {
        return userService.query(queryRequest);
    }

    @PostMapping("save")
    public String save(@RequestBody @Validated UserInput user) {
        return userService.save(user);
    }

    @PostMapping("delete")
    public Boolean delete(@RequestBody List<String> ids) {
        return userService.delete(ids);
    }

    @PostMapping("login")
    public SaTokenInfo login(@RequestBody @Validated UserLoginInput userLoginInput){
       return userService.login(userLoginInput);
    }
}

File: src/apis/__generated/services/UserForFrontController.ts
import type {Executor} from '../';
import type {MenuDto, UserDto} from '../model/dto/';
import type {
    Page, 
    QueryRequest, 
    SaTokenInfo, 
    UserInput, 
    UserLoginInput, 
    UserSpec
} from '../model/static/';

export class UserForFrontController {
    
    constructor(private executor: Executor) {}
    
    delete: (options: UserForFrontControllerOptions['delete']) => Promise<
        boolean | undefined
    > = async(options) => {
        let _uri = '/front/user/delete';
        return (await this.executor({uri: _uri, method: 'POST', body: options.body})) as Promise<boolean | undefined>;
    }
    
    findById: (options: UserForFrontControllerOptions['findById']) => Promise<
        UserDto['UserRepository/USER_ROLE_FETCHER']
    > = async(options) => {
        let _uri = '/front/user/';
        _uri += encodeURIComponent(options.id);
        return (await this.executor({uri: _uri, method: 'GET'})) as Promise<UserDto['UserRepository/USER_ROLE_FETCHER']>;
    }
    
    getUserInfo: () => Promise<
        UserDto['UserRepository/USER_ROLE_FETCHER']
    > = async() => {
        let _uri = '/front/user/info';
        return (await this.executor({uri: _uri, method: 'POST'})) as Promise<UserDto['UserRepository/USER_ROLE_FETCHER']>;
    }
    
    getUserMenus: () => Promise<
        Array<MenuDto['MenuRepository/SIMPLE_FETCHER']>
    > = async() => {
        let _uri = '/front/user/menus';
        return (await this.executor({uri: _uri, method: 'GET'})) as Promise<Array<MenuDto['MenuRepository/SIMPLE_FETCHER']>>;
    }
    
    login: (options: UserForFrontControllerOptions['login']) => Promise<
        SaTokenInfo
    > = async(options) => {
        let _uri = '/front/user/login';
        return (await this.executor({uri: _uri, method: 'POST', body: options.body})) as Promise<SaTokenInfo>;
    }
    
    query: (options: UserForFrontControllerOptions['query']) => Promise<
        Page<UserDto['UserRepository/COMPLEX_FETCHER']>
    > = async(options) => {
        let _uri = '/front/user/query';
        return (await this.executor({uri: _uri, method: 'POST', body: options.body})) as Promise<Page<UserDto['UserRepository/COMPLEX_FETCHER']>>;
    }
    
    save: (options: UserForFrontControllerOptions['save']) => Promise<
        string
    > = async(options) => {
        let _uri = '/front/user/save';
        return (await this.executor({uri: _uri, method: 'POST', body: options.body})) as Promise<string>;
    }
}

export type UserForFrontControllerOptions = {
    'getUserInfo': {}, 
    'findById': {
        id: string
    }, 
    'query': {
        body: QueryRequest<UserSpec>
    }, 
    'save': {
        body: UserInput
    }, 
    'delete': {
        body: Array<string>
    }, 
    'login': {
        body: UserLoginInput
    }, 
    'getUserMenus': {}
}
